fetchcode/package.py

```diff
@@ -0,0 +1,318 @@
+def get_pypi_data_from_purl(purl):
+    """
+    Generate `Package` object from the `purl` string of npm type
+    """
+    purl = PackageURL.from_string(purl)
+    name = purl.name
+    base_path = "https://pypi.org/pypi"
+    api_url = f"{base_path}/{name}/json"
+    response = get_response(api_url)
+    releases = response.get("releases") or {}
+    info = response.get("info") or {}
+    homepage_url = info.get("home_page")
+    license = info.get("license")
+    project_urls = info.get("project_urls") or {}
+    code_view_url = project_urls.get("Source")
+    bug_tracking_url = project_urls.get("Tracker")
+    if not (code_view_url):
+        code_view_url = project_urls.get("Code")
+    if not (bug_tracking_url):
+        bug_tracking_url = project_urls.get("Issue Tracker")
+    if not (code_view_url):
+        code_view_url = project_urls.get("Source Code")
+    if not (bug_tracking_url):
+        bug_tracking_url = project_urls.get("Bug Tracker")
+    yield Package(
+        homepage_url=homepage_url,
+        api_url=api_url,
+        bug_tracking_url=bug_tracking_url,
+        code_view_url=code_view_url,
+        declared_license=license,
+        **purl.to_dict(),
+    )
+    for num in releases:
+        vpurl = PackageURL(type=purl.type, name=name, version=num)
+        release = releases.get(num) or [{}]
+        release = release[0]
+        download_url = release.get("url")
+        yield Package(
+            homepage_url=homepage_url,
+            api_url=api_url,
+            bug_tracking_url=bug_tracking_url,
+            code_view_url=code_view_url,
+            download_url=download_url,
+            declared_license=license,
+            **vpurl.to_dict(),
+        )
+
+
+@router.route("pkg:github/.*")
+def get_github_data_from_purl(purl):
+    """
+    Generate `Package` object from the `purl` string of github type
+    """
+    purl = PackageURL.from_string(purl)
+    name = purl.name
+    namespace = purl.namespace
+    base_path = "https://api.github.com/repos"
+    api_url = f"{base_path}/{namespace}/{name}"
+    response = get_response(api_url)
+    homepage_url = response.get("homepage")
+    vcs_url = response.get("git_url")
+    github_url = "https://github.com"
+    bug_tracking_url = f"{github_url}/{namespace}/{name}/issues"
+    code_view_url = f"{github_url}/{namespace}/{name}"
+    license_data = response.get("license") or {}
+    declared_license = license_data.get("spdx_id")
+    primary_language = response.get("language")
+    yield Package(
+        homepage_url=homepage_url,
+        vcs_url=vcs_url,
+        api_url=api_url,
+        bug_tracking_url=bug_tracking_url,
+        code_view_url=code_view_url,
+        declared_license=declared_license,
+        primary_language=primary_language,
+        **purl.to_dict(),
+    )
+    release_url = f"{api_url}/releases"
+    releases = get_response(release_url)
+    for release in releases:
+        version = release.get("name")
+        vpurl = PackageURL(
```

Comment:
What does `vpurl` mean?

Reply:
Version PURL

Reply:
We should probably make the varible names more descriptive:
```
for release in releases:
    release_name = release.get('name')
    release_purl = PackageURL...
```

Reply:
Yeah thanks for this : 💯

Reply:
I have used `version_purl` for consistency for every package manager, Is it good to go or should I rename it to `release_purl` ?